app/src/main/cpp/skyline/services/timesrv/core.cpp

```diff
@@ -0,0 +1,299 @@
+        TimeSpanType internalOffset{};
+
+        // Setup the standard steady clock from which everything in the system counts
+        managerServer.SetupStandardSteadyClock(rtcId, rtcOffset, internalOffset, {}, false);
+
+        SystemClockContext localSystemClockContext{
+            .timestamp {
+                .timePoint = 0,
+                .clockSourceId = rtcId,
+            },
+            .offset = 0 //!< Zero offset as the RTC is calibrated already
+        };
+        // Don't supply a POSIX time as the offset will be taken from the above context instead.
+        // Normally the POSIX time would be the initial year for the clock to reset to if the context got wiped.
+        managerServer.SetupStandardLocalSystemClock(localSystemClockContext, 0);
+
+        // Use the context just created in local clock for the network clock, HOS gets this from settings
+        auto context{localSystemClock.GetClockContext()};
+        if (!context)
+            throw exception("Failed to get local system clock context!");
+
+        constexpr TimeSpanType sufficientAccuracy{TimeSpanType::FromDays(30)}; //!< https://switchbrew.org/wiki/System_Settings#time
+
+        managerServer.SetupStandardNetworkSystemClock(*context, sufficientAccuracy);
+
+        // Initialise the user system clock with automatic correction disabled as we don't emulate the automatic correction thread
+        managerServer.SetupStandardUserSystemClock(false, SteadyClockTimePoint{.clockSourceId = UUID::GenerateUuidV4()});
+        managerServer.SetupEphemeralSystemClock();
+
+        // Timezone init - normally done in glue
+
+        // Act as if we just updated the current timezone
+        auto timezoneUpdateTime{standardSteadyClock.GetTimePoint()};
+        if (!timezoneUpdateTime)
+            throw exception("Failed to create a timezone updated timepoint!");
+
+        // TODO Checked
+        auto timeZoneBinaryListFile{state.os->assetFileSystem->OpenFile("tzdata/binaryList.txt")};
+        std::vector<u8> buffer(timeZoneBinaryListFile->size);
+        timeZoneBinaryListFile->Read(buffer);
+
+        // Parse binaryList.txt into a vector
+        auto prev{buffer.begin()};
+        for (auto it{buffer.begin()}; it != buffer.end(); it++) {
+            if (*it == '\n' && prev != it) {
+                timesrv::LocationName name{};
+                span(prev.base(), std::distance(prev, std::prev(it))).as_string().copy(name.data(), name.size());
+                locationNameList.push_back(name);
+
+                if (std::next(it) != buffer.end())
+                    prev = std::next(it);
+            }
+        }
+
+        // TODO Checked
+        auto timeZoneBinaryVersionFile{state.os->assetFileSystem->OpenFile("tzdata/version.txt")};
+        std::array<u8, 0x10> timeZoneBinaryVersion{};
+        timeZoneBinaryVersionFile->Read(timeZoneBinaryVersion);
```

Comment:
`tzdata/version.txt` is not `0x10` bytes long which'll lead to `Backing::Read` throwing, you should've tested `time` after rebasing it on `vfs-codestyle`.